bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/container/ModuleResolver.java

```diff
@@ -962,7 +989,20 @@ ModuleResolutionReport resolve() {
 					if (DEBUG_WIRING) {
 						printWirings(result);
 					}
-					report = reportBuilder.build(result, re);
+					int total = logger.totalPerm;
```

Comment:
I would rather have new methods on the report builder to increment the values and then the `build` simply sets the final values into the report directly.  I don't see the need for this extra interface here.

Reply:
done.